app/test_main.py

```diff
@@ -1 +1,56 @@
-# write your code here
+import pytest
+from _pytest.monkeypatch import MonkeyPatch
+from typing import Callable
+from unittest.mock import MagicMock
+from app.main import can_access_google_page
+
+
+@pytest.fixture
+def url() -> str:
+    return ("https://www.google.com/webhp?hl=en&sa="
+            "X&ved=0ahUKEwjlr5aw25eFAxVFFBAIHbjlAdEQPAgJ")
+
+
+@pytest.fixture
+def mock_internet_connection(monkeypatch: MonkeyPatch) -> Callable:
```

Comment:
Can you please explain why did you choose going through all the trouble of making fixtures and using monkey patch instead of making mock in one line of code in your test function?